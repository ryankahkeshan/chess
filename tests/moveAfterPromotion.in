setup
+ K e1 + k e8
+ R a1
+ P c7
+ p h3
done game human human

move c7 c8 r

move c7 c8 P
move c7 c8 R
move e8 f7
move c8 c3
move f7 e8
move a1 a8
move e8 f7

move e1 g1

move e1 c1
move e1 f1
move h3 h2

move f1 g1
move f1 g2

move h2 h1 p
move h2 h1 B
move h2 h1 b
resign
