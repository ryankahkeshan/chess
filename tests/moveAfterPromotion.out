8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8  _ _k_ _
7 _ P _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8  _ _k_ _
7 _ P _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8  _ _k_ _
7 _ P _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

Invalid move
Invalid move
8  _R_k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

Black is in check.
8  _R_ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8  _ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _p
2  _ _ _ _
1 R _ K _ 

  abcdefgh

8 R_ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _p
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black is in check.
8 R_ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _p
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid move
Invalid move
8 R_ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _p
2  _ _ _ _
1 _ _ _K_ 

  abcdefgh

8 R_ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _ 
2  _ _ _ p
1 _ _ _K_ 

  abcdefgh

8 R_ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _ 
2  _ _ _ p
1 _ _ _ K 

  abcdefgh

Invalid move
Invalid move
Invalid move
8 R_ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ R _ _ 
2  _ _ _ _
1 _ _ _ Kb

  abcdefgh

White is in check.
Black wins!
Final Score:
White: 0
Black: 1
